src/stash/src/lib.rs

```diff
@@ -724,6 +724,54 @@ where
             .await
     }
 
+    /// Transactionally deletes any kv pair from `self` whose key is in `keys`.
+    ///
+    /// Note that:
+    /// - Unlike `delete`, this operation operates in time O(keys), and not
+    ///   O(set).
+    /// - This operation runs in a single transaction and cannot be combined
+    ///   with other transactions.
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn delete_keys(&self, stash: &mut Stash, keys: BTreeSet<K>) -> Result<(), StashError>
+    where
+        K: Hash + Clone + 'static,
+        V: Clone,
+    {
+        let name = self.name;
+        stash
+            .with_transaction(move |tx| {
+                Box::pin(async move {
+                    let collection = tx.collection::<K, V>(name).await?;
+                    let lower = tx.upper(collection.id).await?;
+                    let mut batch = collection.make_batch_lower(lower)?;
+                    let set = match tx.peek_one(collection).await {
```

Comment:
The mean number of things we're deleting is likely 2. No need to over-engineer this because we don't know that this is actually a problem; I can just use the existing `delete` API. tftr!

Reply:
If it's 2 then might be worth doing the peek_key_one version even if it's not pipelined.

Reply:
Took a stab at this; lmk what you think

Reply:
LGTM